Remotion/Core/Core/Context/CallContextStorageProvider.obsolete.cs

```diff
@@ -16,33 +16,28 @@
 // 
 using System;
 using System.Runtime.Remoting.Messaging;
-using Remotion.ServiceLocation;
-using Remotion.Utilities;
 
 namespace Remotion.Context
 {
   /// <summary>
   /// Implements <see cref="ISafeContextStorageProvider"/> by storing data in the thread-local <see cref="CallContext"/>.
   /// </summary>
-  [ImplementationFor (typeof (ISafeContextStorageProvider), Position = 1)]
+  [Obsolete ("CallContext is deprecated in .NET Core. Use AsyncLocalStorageProvider instead. (Version 3.0.0-alpha.14)", true)]
   public class CallContextStorageProvider : ISafeContextStorageProvider
   {
     public object GetData (string key)
     {
-      ArgumentUtility.CheckNotNull ("key", key);
-      return CallContext.GetData (key);
+      throw new NotSupportedException();
```

Comment:
Please add the obsolete message from the attribute to the exceptions. This covers the odd case when there's been no re-compile.